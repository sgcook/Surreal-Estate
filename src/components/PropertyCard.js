import React from "react";
import PropTypes from "prop-types";
import "../styles/property-card.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faStar,
  faBath,
  faBed,
  faEnvelope,
} from "@fortawesome/free-solid-svg-icons";

const PropertyCard = ({
  _id,
  bathrooms,
  bedrooms,
  city,
  email,
  price,
  title,
  type,
  userID,
  onSaveProperty,
}) => {
  return (
    <div className="property-card">
      <h3>{title}</h3>
      <ul className="property-card__list">
        <li className="type-city">
          {type} - {city}
        </li>
        <li className="bathrooms">
          <FontAwesomeIcon className="icon" icon={faBath} />
          {bathrooms}
        </li>
        <li className="bedrooms">
          <FontAwesomeIcon className="icon" icon={faBed} />
          {bedrooms}
        </li>
        <li className="price">Â£{price}</li>
      </ul>
      <p className="email-p">
        <a href={`mailto:${email}`} className="email-link">
          <FontAwesomeIcon className="icon" icon={faEnvelope} />
          Email
        </a>
      </p>
      {userID && (
        <button
          type="button"
          className="save"
          onClick={() => onSaveProperty(_id)}
        >
          <FontAwesomeIcon className="icon" icon={faStar} />
          Save
        </button>
      )}
    </div>
  );
};

PropertyCard.propTypes = {
  _id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bathrooms: PropTypes.string.isRequired,
  bedrooms: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  userID: PropTypes.string,
  onSaveProperty: PropTypes.func,
};

PropertyCard.defaultProps = {
  userID: "",
  onSaveProperty: () => {},
};

export default PropertyCard;
