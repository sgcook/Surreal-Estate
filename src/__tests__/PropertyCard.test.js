import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    bathrooms: 3,
    bedrooms: 2,
    city: "Leeds",
    email: "joe@gmail.com",
    price: "20000.00",
    title: "beautiful home",
    type: "bungalow",
  };

  test("renders PropertyCard", () => {
    const { asFragment } = render(
      <PropertyCard
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        city={validProps.city}
        email={validProps.email}
        price={validProps.price}
        title={validProps.title}
        type={validProps.type}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders bathroom prop correctly", () => {
    const { getByText } = render(
      <PropertyCard
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        city={validProps.city}
        email={validProps.email}
        price={validProps.price}
        title={validProps.title}
        type={validProps.type}
      />
    );

    expect(getByText("3")).toHaveClass("bathrooms");
  });

  test("renders bedroom prop correctly", () => {
    const { getByText } = render(
      <PropertyCard
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        city={validProps.city}
        email={validProps.email}
        price={validProps.price}
        title={validProps.title}
        type={validProps.type}
      />
    );

    expect(getByText("2")).toHaveClass("bedrooms");
  });

  test("renders email prop correctly", () => {
    const { getByText } = render(
      <PropertyCard
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        city={validProps.city}
        email={validProps.email}
        price={validProps.price}
        title={validProps.title}
        type={validProps.type}
      />
    );

    expect(getByText("Email")).toHaveAttribute("href", "mailto:joe@gmail.com");
  });

  test("renders price prop correctly", () => {
    const { getByText } = render(
      <PropertyCard
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        city={validProps.city}
        email={validProps.email}
        price={validProps.price}
        title={validProps.title}
        type={validProps.type}
      />
    );

    expect(getByText("Â£20000")).toHaveClass("price");
  });

  test("renders city and type props correctly", () => {
    const { getByText } = render(
      <PropertyCard
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        city={validProps.city}
        email={validProps.email}
        price={validProps.price}
        title={validProps.title}
        type={validProps.type}
      />
    );

    expect(getByText("bungalow - Leeds")).toBeInstanceOf(HTMLLIElement);
  });

  test("renders title prop correctly", () => {
    const { getByText } = render(
      <PropertyCard
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        city={validProps.city}
        email={validProps.email}
        price={validProps.price}
        title={validProps.title}
        type={validProps.type}
      />
    );

    expect(getByText("beautiful home")).toBeInstanceOf(HTMLHeadingElement);
  });
});
