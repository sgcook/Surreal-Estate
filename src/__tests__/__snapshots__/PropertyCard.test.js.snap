// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders PropertyCard 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    I am a property card
    <h3>
      beautiful home
    </h3>
    <ul
      class="property-card__list"
    >
      <li
        class="type-ctiy"
      >
        bungalow - Leeds
      </li>
      <li
        class="bathrooms"
      >
        3
      </li>
      <li
        class="bedrooms"
      >
        2
      </li>
      <li
        class="price"
      >
        Â£20000
      </li>
    </ul>
    <p
      class="email-p"
    >
      <a
        class="email-link"
        href="mailto:joe@gmail.com"
      >
        Email
      </a>
    </p>
  </div>
</DocumentFragment>
`;
